actions/microsoft-calendar/actions.py

```diff
@@ -0,0 +1,213 @@
+            "attendees", []
+        )
+
+    response = requests.patch(
+        f"https://graph.microsoft.com/v1.0/me/events/{event_id}",
+        headers=headers,
+        json=updates_json,
+    )
+
+    if response.status_code != 200:
+        return Response(error=response.text)
+
+    return Response(result=Event.model_validate(response.json()))
+
+
+@action(is_consequential=False)
+def list_events(
+    credentials: OAuth2Secret[
+        Literal["microsoft"],
+        list[Literal["Calendars.Read"]],
+    ],
+    query_params: QueryParams,
+    timezone: str,
+    calendar_id: str = "",
+) -> Response[list[Event]]:
+    """List all events in the user's calendar.
+
+    To aggregate all events across calendars, call this method for each calendar returned by list_calendars endpoint.
+    Ideally a start and end date should always be provided, otherwise we will look in the whole future or past.
+
+    Args:
+        credentials: JSON containing Microsoft OAuth2 credentials.
+        query_params: JSON containing the following query parameters:
+            - filter -> used to filter results based on different properties
+                Details available at: https://learn.microsoft.com/en-us/graph/filter-query-parameter?tabs=http.
+                Usage example: "start/dateTime ge '2024-08-07T00:00:00Z' and start/dateTime lt '2024-08-07T23:59:59Z'".
+            - search -> used to search after a particular string
+        timezone: By default, it should match the value returned by the get_mailbox_timezone action unless specified
+            otherwise by the user. Queries and event dates will use the timezone specified.
+        calendar_id: Calendar identifier which can be found by listing calendars action.
+            Default value is using the user's default calendar.
+
+    Returns:
+        A list of calendar events that match the query, if defined.
+
+    """
+    url = "https://graph.microsoft.com/v1.0/me/calendar/events"
+    if calendar_id:
+        url = f"https://graph.microsoft.com/v1.0/me/calendars/{calendar_id}/events"
+
+    response = requests.get(
+        url,
+        headers=_build_headers(credentials, timezone=timezone),
+        params=query_params.model_dump(by_alias=True, exclude_none=True),
+    )
+
+    if response.status_code != 200:
+        return Response(error=response.text)
+
+    events = [Event.model_validate(event) for event in response.json()["value"]]
+
+    return Response(result=events)
+
+
+@action(is_consequential=False)
+def list_calendars(
+    credentials: OAuth2Secret[
+        Literal["microsoft"],
+        list[Literal["Calendars.Read"]],
+    ],
+) -> Response[list[Calendar]]:
+    """List all calendars that the user is subscribed to.
+
+    Args:
+        credentials: JSON containing Microsoft OAuth2 credentials.
+
+    Returns:
+        A list of calendars.
+
+    """
+    response = requests.get(
+        "https://graph.microsoft.com/v1.0/me/calendars",
```

Comment:
I think these should be transitioned to constants. 
That would help with a better management for the future API versions or changes.